ql/src/java/org/apache/hadoop/hive/ql/ppd/SyntheticJoinPredicate.java

```diff
@@ -339,6 +339,12 @@ String getFuncText(String funcText, final int srcPos) {
           vector.add(right, left);
           break;
         case JoinDesc.LEFT_OUTER_JOIN:
+        case JoinDesc.ANTI_JOIN:
+        //TODO : In case of anti join, bloom filter can be created on left side also ("IN (keylist right table)").
+        // But the filter should be "not-in" ("NOT IN (keylist right table)") as we want to select the records from
+        // left side which are not present in the right side. But it may cause wrong result as
+        // bloom filter may have false positive and thus simply adding not is not correct,
+        // special handling is required for "NOT IN".
```

Comment:
Makes sense, for this particular purpose in the future we could something like ``The opossite bloom filter`` to support such cases 
https://github.com/jmhodges/opposite_of_a_bloom_filter/

Reply:
created a Jira ..https://issues.apache.org/jira/browse/HIVE-23903

Reply:
Thank Mahesh! Has this in the back of my head for a while -- this will be useful for a bunch of cases including anti-joins